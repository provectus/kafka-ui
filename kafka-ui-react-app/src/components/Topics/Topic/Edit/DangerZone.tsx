import { ErrorMessage } from '@hookform/error-message';
import ConfirmationModal from 'components/common/ConfirmationModal/ConfirmationModal';
import React from 'react';
import { useForm } from 'react-hook-form';

export interface Props {
  clusterName: string;
  topicName: string;
  defaultPartitions: number;
  defaultReplicationFactor: number;
  partitionsCountIncreased: boolean;
  replicationFactorUpdated: boolean;
  updateTopicPartitionsCount: (
    clusterName: string,
    topicname: string,
    partitions: number
  ) => void;
  updateTopicReplicationFactor: (
    clusterName: string,
    topicname: string,
    replicationFactor: number
  ) => void;
}

const DangerZone: React.FC<Props> = ({
  clusterName,
  topicName,
  defaultPartitions,
  defaultReplicationFactor,
  partitionsCountIncreased,
  replicationFactorUpdated,
  updateTopicPartitionsCount,
  updateTopicReplicationFactor,
}) => {
  const [isPartitionsConfirmationVisible, setIsPartitionsConfirmationVisible] =
    React.useState<boolean>(false);
  const [
    isReplicationFactorConfirmationVisible,
    setIsReplicationFactorConfirmationVisible,
  ] = React.useState<boolean>(false);
  const [partitions, setPartitions] = React.useState<number>(defaultPartitions);
  const [replicationFactor, setReplicationFactor] = React.useState<number>(
    defaultReplicationFactor
  );

  const {
    register: partitionsRegister,
    handleSubmit: handlePartitionsSubmit,
    formState: partitionsFormState,
    setError: setPartitionsError,
    getValues: partitionsGetValues,
  } = useForm({
    defaultValues: {
      partitions,
    },
  });

  const {
    register: replicationFactorRegister,
    handleSubmit: handleКeplicationFactorSubmit,
    formState: replicationFactorFormState,
    getValues: replicationFactorgetValues,
  } = useForm({
    defaultValues: {
      replicationFactor,
    },
  });

  const validatePartitions = (data: { partitions: number }) => {
    if (data.partitions < defaultPartitions) {
      setPartitionsError('partitions', {
        type: 'manual',
        message: 'You can only increase the number of partitions!',
      });
    } else {
      setPartitions(data.partitions);
      setIsPartitionsConfirmationVisible(true);
    }
  };

  const validateReplicationFactor = (data: { replicationFactor: number }) => {
    setReplicationFactor(data.replicationFactor);
    setIsReplicationFactorConfirmationVisible(true);
  };

  React.useEffect(() => {
    if (partitionsCountIncreased) {
      setIsPartitionsConfirmationVisible(false);
    }
  }, [partitionsCountIncreased]);

  React.useEffect(() => {
    if (replicationFactorUpdated) {
      setIsReplicationFactorConfirmationVisible(false);
    }
  }, [replicationFactorUpdated]);

  const partitionsSubmit = () => {
    updateTopicPartitionsCount(
      clusterName,
      topicName,
      partitionsGetValues('partitions')
    );
  };
  const replicationFactorSubmit = () => {
    updateTopicReplicationFactor(
      clusterName,
      topicName,
      replicationFactorgetValues('replicationFactor')
    );
  };
  return (
    <div className="box">
      <h4 className="title is-5 has-text-danger mb-5">Danger Zone</h4>
      <div className="is-flex is-flex-direction-column">
        <form
          onSubmit={handlePartitionsSubmit(validatePartitions)}
          className="columns mb-0"
        >
          <div className="column is-three-quarters">
            <label className="label">Number of partitions *</label>
            <input
              className="input"
              type="number"
              placeholder="Number of partitions"
              {...partitionsRegister('partitions', {
                required: 'Partiotions are required',
              })}
            />
          </div>
          <div className="column is-flex is-align-items-flex-end">
            <input
              type="submit"
              className="button is-danger"
              disabled={!partitionsFormState.isDirty}
            />
          </div>
        </form>
        <p className="help is-danger mt-0 mb-4">
          <ErrorMessage errors={partitionsFormState.errors} name="partitions" />
        </p>
        <ConfirmationModal
          isOpen={isPartitionsConfirmationVisible}
          onCancel={() => setIsPartitionsConfirmationVisible(false)}
          onConfirm={partitionsSubmit}
        >
          Are you sure you want to increase the number of partitions? Do it only
          if you 100% know what you are doing!
        </ConfirmationModal>

        <form
          onSubmit={handleКeplicationFactorSubmit(validateReplicationFactor)}
          className="columns"
        >
          <div className="column is-three-quarters">
            <label className="label">Replication Factor *</label>
            <input
              className="input"
              type="number"
              placeholder="Replication Factor"
              {...replicationFactorRegister('replicationFactor', {
                required: 'Replication Factor are required',
              })}
            />
          </div>
          <div className="column is-flex is-align-items-flex-end">
            <input
              type="submit"
              className="button is-danger"
              disabled={!replicationFactorFormState.isDirty}
            />
          </div>
        </form>
        <p className="help is-danger mt-0">
          <ErrorMessage
            errors={replicationFactorFormState.errors}
            name="replicationFactor"
          />
        </p>
        <ConfirmationModal
          isOpen={isReplicationFactorConfirmationVisible}
          onCancel={() => setIsReplicationFactorConfirmationVisible(false)}
          onConfirm={replicationFactorSubmit}
        >
          Are you sure you want to update the replication factor?
        </ConfirmationModal>
      </div>
    </div>
  );
};

export default DangerZone;
