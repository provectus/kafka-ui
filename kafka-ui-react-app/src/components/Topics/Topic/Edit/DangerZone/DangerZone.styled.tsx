import styled from 'styled-components';

export const Wrapper = styled.div`
  margin-top: 16px;
  padding: 16px;
  border: 1px solid ${({ theme }) => theme.dangerZone.borderColor};
  box-sizing: border-box;
  width: 768px;

  & > div {
    display: flex;
    flex-direction: column;
    gap: 8px;
  }
`;

export const Title = styled.h1`
  color: ${({ theme }) => theme.dangerZone.color.title};
  font-size: 20px;
  padding-bottom: 16px;
`;
export const Warning = styled.div`
  color: ${({ theme }) => theme.dangerZone.color.warningMessage};
  font-size: 12px;
  padding-bottom: 16px;
`;
export const Form = styled.form`
  display: flex;
  align-items: flex-end;
  gap: 16px;
  & > *:first-child {
    flex-grow: 4;
  }
  & > *:last-child {
    flex-grow: 1;
  }
`;
