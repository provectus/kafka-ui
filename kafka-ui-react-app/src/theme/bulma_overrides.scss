.has {
  &-text-overflow-ellipsis {
    flex: 1;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  &-text-nowrap {
    white-space: nowrap;
  }

  &-content-overflow-ellipsis {
    max-height: 73px;
    overflow: hidden;
    text-overflow: ellipsis;
    background-color: #fff;

    background: -webkit-linear-gradient(
      90deg,
      rgba(0, 0, 0, 0.1) 0%,
      rgba(0, 0, 0, 1) 40%
    );
    background-clip: text;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }
}

.breadcrumb li {
  &.is-active > span {
    padding: 0 0.75em;
  }

  &:first-child > span {
    padding-left: 0;
  }
}

.section {
  animation: fadein 0.5s;
}

.select.is-block select {
  width: 100%;
}

.notification {
  &.is-light {
    &.is-primary {
      background-color: #ebfffc;
      color: #00947e;
    }

    &.is-danger {
      background-color: #feecf0;
      color: #cc0f35;
    }
  }
}

.is-family-code {
  font-family: 'Roboto Mono', sans-serif !important;
  font-size: 0.9rem;
  line-height: 1.5em;
}

@keyframes fadein {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

.level.level-multiline {
  flex-wrap: wrap;
  .level-item.is-one-third {
    flex-basis: 26%;
  }
  .level-item.is-one-third:nth-child(n + 4) {
    margin-top: 10px;
  }
}

.is-size-8 {
  font-size: $size-8;
}

.box {
  box-shadow: none;
}

.tag:not(body) {
  font-size: 0.75rem;
  height: 1.75em;
  line-height: 1.75;
}
