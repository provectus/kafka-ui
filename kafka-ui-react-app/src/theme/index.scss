@import 'src/theme/bulma_overrides';

#root, body, html {
  width: 100%;
  position: relative;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  background: repeating-linear-gradient(
    145deg,
    rgba(0,0,0,.003),
    rgba(0,0,0,.005) 5px,
    rgba(0,0,0,0) 5px,
    rgba(0,0,0,0) 10px
  ),
  repeating-linear-gradient(
    -145deg,
    rgba(0,0,0,.003),
    rgba(0,0,0,.005) 5px,
    rgba(0,0,0,0) 5px,
    rgba(0,0,0,0) 10px
  );
  background-color: $light;

}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.cursor-pointer {
  cursor: pointer;
}
