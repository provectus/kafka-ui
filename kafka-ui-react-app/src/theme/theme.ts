/* eslint-disable import/prefer-default-export */
export const Colors = {
  neutral: {
    '0': '#FFFFFF',
    '5': '#F1F2F3',
    '10': '#E3E6E8',
    '15': '#D5DADD',
    '20': '#C7CED1',
    '30': '#ABB5BA',
    '40': '#8F9CA3',
    '50': '#73848C',
    '60': '#5C6970',
    '70': '#454F54',
    '80': '#2F3639',
    '90': '#171A1C',
  },
  green: {
    '10': '#D6F5E0',
    '15': '#C2F0D1',
    '30': '#85E0A3',
    '40': '#5CD685',
    '60': '#29A352',
  },
  brand: {
    '20': '#A3A3F5',
    '50': '#4F4FFF',
    '55': '#1717CF',
    '60': '#1414B8',
  },
  red: {
    '10': '#FAD1D1',
    '50': '#E51A1A',
  },
  yellow: {
    '10': '#FFEECC',
  },
};

const theme = {
  layout: {
    minWidth: '1200px',
    navBarWidth: '201px',
    navBarHeight: '3.25rem',
  },
  alert: {
    color: {
      error: Colors.red[10],
      success: Colors.green[10],
      warning: Colors.yellow[10],
      info: Colors.neutral[10],
    },
  },
  buttonStyles: {
    primary: {
      backgroundColor: {
        normal: Colors.brand[50],
        hover: Colors.brand[55],
        active: Colors.brand[60],
      },
      color: Colors.neutral[0],
      invertedColors: {
        normal: Colors.brand[50],
        hover: Colors.brand[55],
        active: Colors.brand[60],
      },
    },
    secondary: {
      backgroundColor: {
        normal: Colors.neutral[5],
        hover: Colors.neutral[10],
        active: Colors.neutral[15],
      },
      color: Colors.neutral[90],
      invertedColors: {
        normal: Colors.neutral[50],
        hover: Colors.neutral[70],
        active: Colors.neutral[90],
      },
    },
    height: {
      S: '24px',
      M: '32px',
      L: '40px',
    },
    fontSize: {
      S: '14px',
      M: '14px',
      L: '16px',
    },
  },
  menuStyles: {
    backgroundColor: {
      normal: Colors.neutral[0],
      hover: Colors.neutral[5],
      active: Colors.neutral[10],
    },
    color: {
      normal: Colors.neutral[50],
      hover: Colors.neutral[50],
      active: Colors.neutral[90],
    },
    statusIconColor: {
      online: Colors.green[40],
      offline: Colors.red[50],
    },
    chevronIconColor: Colors.neutral[50],
  },
  thStyles: {
    backgroundColor: {
      normal: Colors.neutral[0],
    },
    color: {
      normal: Colors.neutral[50],
    },
    previewColor: {
      normal: Colors.brand[50],
    },
  },
  primaryTabStyles: {
    color: {
      normal: Colors.neutral[50],
      hover: Colors.neutral[90],
      active: Colors.neutral[90],
    },
    borderColor: {
      normal: 'transparent',
      hover: 'transparent',
      active: Colors.brand[50],
    },
  },
  secondaryTabStyles: {
    backgroundColor: {
      normal: Colors.neutral[0],
      hover: Colors.neutral[5],
      active: Colors.neutral[10],
    },
    color: {
      normal: Colors.neutral[50],
      hover: Colors.neutral[90],
      active: Colors.neutral[90],
    },
  },
  selectStyles: {
    color: {
      normal: Colors.neutral[90],
      hover: Colors.neutral[90],
      active: Colors.neutral[90],
      disabled: Colors.neutral[30],
    },
    borderColor: {
      normal: Colors.neutral[30],
      hover: Colors.neutral[50],
      active: Colors.neutral[70],
      disabled: Colors.neutral[10],
    },
  },
  tagStyles: {
    backgroundColor: {
      green: Colors.green[10],
      gray: Colors.neutral[10],
      yellow: Colors.yellow[10],
    },
    color: Colors.neutral[90],
  },
  paginationStyles: {
    borderColor: {
      normal: Colors.neutral[30],
      hover: Colors.neutral[50],
      active: Colors.neutral[70],
      disabled: Colors.neutral[20],
    },
    color: {
      normal: Colors.neutral[90],
      hover: Colors.neutral[90],
      active: Colors.neutral[90],
      disabled: Colors.neutral[20],
    },
  },
  switch: {
    unchecked: Colors.neutral[30],
    checked: Colors.green[60],
  },
  pageLoader: {
    borderColor: Colors.brand[50],
    borderBottomColor: Colors.neutral[0],
  },
  metrics: {
    backgroundColor: Colors.neutral[5],
    indicator: {
      backgroundColor: Colors.neutral[0],
      titleColor: Colors.neutral[50],
      warningTextColor: Colors.red[50],
      lightTextColor: Colors.neutral[30],
    },
  },
};

export type ThemeType = typeof theme;

export default theme;
