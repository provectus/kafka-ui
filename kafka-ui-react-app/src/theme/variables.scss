// Typography
$size-7: 0.875rem;
$size-8: 0.75rem;
$body-line-height: 1.34;
$body-color: hsl(0, 100%, 0%);

// Section
$section-padding-desktop: 0 0 0.5rem;
$section-padding: 1rem 1.5rem 3rem;

// Tag
$tag-radius: 16px;
