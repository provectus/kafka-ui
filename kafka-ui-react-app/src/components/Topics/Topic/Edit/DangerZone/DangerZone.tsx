import { ErrorMessage } from '@hookform/error-message';
import { Button } from 'components/common/Button/Button';
import Input from 'components/common/Input/Input';
import { FormError } from 'components/common/Input/Input.styled';
import { InputLabel } from 'components/common/Input/InputLabel.styled';
import React from 'react';
import { FormProvider, useForm } from 'react-hook-form';
import { RouteParamsClusterTopic } from 'lib/paths';
import useAppParams from 'lib/hooks/useAppParams';
import { useConfirm } from 'lib/hooks/useConfirm';
import {
  useIncreaseTopicPartitionsCount,
  useUpdateTopicReplicationFactor,
} from 'lib/hooks/api/topics';

import * as S from './DangerZone.styled';

export interface DangerZoneProps {
  defaultPartitions: number;
  defaultReplicationFactor: number;
}

const DangerZone: React.FC<DangerZoneProps> = ({
  defaultPartitions,
  defaultReplicationFactor,
}) => {
  const params = useAppParams<RouteParamsClusterTopic>();
  const [partitions, setPartitions] = React.useState<number>(defaultPartitions);
  const [replicationFactor, setReplicationFactor] = React.useState<number>(
    defaultReplicationFactor
  );
  const increaseTopicPartitionsCount = useIncreaseTopicPartitionsCount(params);
  const updateTopicReplicationFactor = useUpdateTopicReplicationFactor(params);

  const partitionsMethods = useForm({
    defaultValues: {
      partitions,
    },
  });

  const replicationFactorMethods = useForm({
    defaultValues: {
      replicationFactor,
    },
  });

  const confirm = useConfirm();
  const confirmPartitionsChange = () =>
    confirm(
      `Are you sure you want to increase the number of partitions?
        Do it only if you 100% know what you are doing!`,
      () =>
        increaseTopicPartitionsCount.mutateAsync(
          partitionsMethods.getValues('partitions')
        )
    );
  const confirmReplicationFactorChange = () =>
    confirm('Are you sure you want to update the replication factor?', () =>
      updateTopicReplicationFactor.mutateAsync(
        replicationFactorMethods.getValues('replicationFactor')
      )
    );

  const validatePartitions = (data: { partitions: number }) => {
    if (data.partitions < defaultPartitions) {
      partitionsMethods.setError('partitions', {
        type: 'manual',
        message: 'You can only increase the number of partitions!',
      });
    } else {
      setPartitions(data.partitions);
      confirmPartitionsChange();
    }
  };

  const validateReplicationFactor = (data: { replicationFactor: number }) => {
    try {
      setReplicationFactor(data.replicationFactor);
      confirmReplicationFactorChange();
    } catch (e) {
      // do nothing
    }
  };

  return (
    <S.Wrapper>
      <S.Title>Danger Zone</S.Title>
      <S.Warning>
        Change these parameters only if you are absolutely sure what you are
        doing.
      </S.Warning>
      <div>
        <FormProvider {...partitionsMethods}>
          <S.Form
            onSubmit={partitionsMethods.handleSubmit(validatePartitions)}
            aria-label="Edit number of partitions"
          >
            <div>
              <InputLabel htmlFor="partitions">
                Number of partitions *
              </InputLabel>
              <Input
                inputSize="M"
                type="number"
                id="partitions"
                name="partitions"
                hookFormOptions={{
                  required: 'Partiotions are required',
                }}
                placeholder="Number of partitions"
              />
            </div>
            <div>
              <Button
                buttonType="primary"
                buttonSize="M"
                type="submit"
                disabled={!partitionsMethods.formState.isDirty}
              >
                Submit
              </Button>
            </div>
          </S.Form>
        </FormProvider>
        <FormError>
          <ErrorMessage
            errors={partitionsMethods.formState.errors}
            name="partitions"
          />
        </FormError>
        <FormProvider {...replicationFactorMethods}>
          <S.Form
            onSubmit={replicationFactorMethods.handleSubmit(
              validateReplicationFactor
            )}
            aria-label="Edit replication factor"
          >
            <div>
              <InputLabel htmlFor="replicationFactor">
                Replication Factor *
              </InputLabel>
              <Input
                id="replicationFactor"
                inputSize="M"
                type="number"
                placeholder="Replication Factor"
                name="replicationFactor"
                hookFormOptions={{
                  required: 'Replication Factor are required',
                }}
              />
            </div>
            <div>
              <Button
                buttonType="primary"
                buttonSize="M"
                type="submit"
                disabled={!replicationFactorMethods.formState.isDirty}
              >
                Submit
              </Button>
            </div>
          </S.Form>
        </FormProvider>
        <FormError>
          <ErrorMessage
            errors={replicationFactorMethods.formState.errors}
            name="replicationFactor"
          />
        </FormError>
      </div>
    </S.Wrapper>
  );
};

export default DangerZone;
