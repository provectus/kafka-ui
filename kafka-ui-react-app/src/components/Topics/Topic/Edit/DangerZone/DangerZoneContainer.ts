import { connect } from 'react-redux';
import { RootState } from 'redux/interfaces';
import {
  updateTopicPartitionsCount,
  updateTopicReplicationFactor,
} from 'redux/reducers/topics/topicsSlice';
import {
  getTopicPartitionsCountIncreased,
  getTopicReplicationFactorUpdated,
} from 'redux/reducers/topics/selectors';

import DangerZone from './DangerZone';

type OwnProps = {
  defaultPartitions: number;
  defaultReplicationFactor: number;
};

const mapStateToProps = (
  state: RootState,
  { defaultPartitions, defaultReplicationFactor }: OwnProps
) => ({
  defaultPartitions,
  defaultReplicationFactor,
  partitionsCountIncreased: getTopicPartitionsCountIncreased(state),
  replicationFactorUpdated: getTopicReplicationFactorUpdated(state),
});

const mapDispatchToProps = {
  updateTopicPartitionsCount,
  updateTopicReplicationFactor,
};

export default connect(mapStateToProps, mapDispatchToProps)(DangerZone);
