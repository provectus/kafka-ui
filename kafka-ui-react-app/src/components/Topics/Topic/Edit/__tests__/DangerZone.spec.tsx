import React from 'react';
import DangerZone, {
  Props,
} from 'components/Topics/Topic/Edit/DangerZone/DangerZone';
import { render, screen, waitFor } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from 'theme/theme';
import userEvent from '@testing-library/user-event';

const setupWrapper = (props?: Partial<Props>) => (
  <ThemeProvider theme={theme}>
    <DangerZone
      clusterName="testCluster"
      topicName="testTopic"
      defaultPartitions={3}
      defaultReplicationFactor={3}
      partitionsCountIncreased={false}
      replicationFactorUpdated={false}
      updateTopicPartitionsCount={jest.fn()}
      updateTopicReplicationFactor={jest.fn()}
      {...props}
    />
  </ThemeProvider>
);

describe('DangerZone', () => {
  it('is rendered properly', () => {
    const component = render(setupWrapper());
    expect(component.baseElement).toMatchSnapshot();
  });

  it('calls updateTopicPartitionsCount', async () => {
    const mockUpdateTopicPartitionsCount = jest.fn();
    render(
      setupWrapper({
        updateTopicPartitionsCount: mockUpdateTopicPartitionsCount,
      })
    );

    userEvent.type(screen.getByLabelText('Number of partitions *'), '4');
    userEvent.click(screen.getByTestId('partitionsSubmit'));

    await waitFor(() => {
      userEvent.click(screen.getAllByText('Submit')[1]);
      expect(mockUpdateTopicPartitionsCount).toHaveBeenCalledTimes(1);
    });
  });

  it('calls updateTopicReplicationFactor', async () => {
    const mockUpdateTopicReplicationFactor = jest.fn();
    render(
      setupWrapper({
        updateTopicReplicationFactor: mockUpdateTopicReplicationFactor,
      })
    );

    userEvent.type(screen.getByLabelText('Replication Factor *'), '4');
    userEvent.click(screen.getByTestId('replicationFactorSubmit'));
    await waitFor(() => {
      userEvent.click(screen.getAllByText('Submit')[2]);
    });

    await waitFor(() => {
      expect(mockUpdateTopicReplicationFactor).toHaveBeenCalledTimes(1);
    });
  });
});
