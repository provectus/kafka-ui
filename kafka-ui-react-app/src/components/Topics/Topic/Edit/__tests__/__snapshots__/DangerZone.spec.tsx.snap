// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DangerZone is rendered properly 1`] = `
.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 12px;
  border: none;
  border-radius: 4px;
  white-space: nowrap;
  background: #4F4FFF;
  color: #FFFFFF;
  font-size: 14px;
  height: 32px;
}

.c6:hover:enabled {
  background: #1717CF;
  color: #FFFFFF;
  cursor: pointer;
}

.c6:active:enabled {
  background: #1414B8;
  color: #FFFFFF;
}

.c6:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c6 a {
  color: white;
}

.c6 i {
  margin-right: 7px;
}

.c5 {
  border: 1px #ABB5BA solid;
  border-radius: 4px;
  height: 32px;
  width: 100%;
  padding-left: 12px;
  font-size: 14px;
}

.c5::-webkit-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c5::-moz-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c5:-ms-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c5::placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c5:hover {
  border-color: #73848C;
}

.c5:focus {
  outline: none;
  border-color: #454F54;
}

.c5:focus::-webkit-input-placeholder {
  color: transparent;
}

.c5:focus::-moz-placeholder {
  color: transparent;
}

.c5:focus:-ms-input-placeholder {
  color: transparent;
}

.c5:focus::placeholder {
  color: transparent;
}

.c5:disabled {
  color: #ABB5BA;
  border-color: #E3E6E8;
  cursor: not-allowed;
}

.c5:read-only {
  color: #171A1C;
  border: none;
  background-color: #F1F2F3;
  cursor: not-allowed;
}

.c5:-moz-read-only:focus::placeholder {
  color: #ABB5BA;
}

.c5:read-only:focus::placeholder {
  color: #ABB5BA;
}

.c8 {
  border: 1px #ABB5BA solid;
  border-radius: 4px;
  height: 40px;
  width: 100%;
  padding-left: 12px;
  font-size: 14px;
}

.c8::-webkit-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8::-moz-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8:-ms-input-placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8::placeholder {
  color: #ABB5BA;
  font-size: 14px;
}

.c8:hover {
  border-color: #73848C;
}

.c8:focus {
  outline: none;
  border-color: #454F54;
}

.c8:focus::-webkit-input-placeholder {
  color: transparent;
}

.c8:focus::-moz-placeholder {
  color: transparent;
}

.c8:focus:-ms-input-placeholder {
  color: transparent;
}

.c8:focus::placeholder {
  color: transparent;
}

.c8:disabled {
  color: #ABB5BA;
  border-color: #E3E6E8;
  cursor: not-allowed;
}

.c8:read-only {
  color: #171A1C;
  border: none;
  background-color: #F1F2F3;
  cursor: not-allowed;
}

.c8:-moz-read-only:focus::placeholder {
  color: #ABB5BA;
}

.c8:read-only:focus::placeholder {
  color: #ABB5BA;
}

.c7 {
  color: #E51A1A;
  font-size: 12px;
}

.c4 {
  position: relative;
}

.c3 {
  font-weight: 500;
  font-size: 12px;
  line-height: 20px;
  color: #454F54;
}

.c0 {
  margin-top: 16px;
  padding: 8px 16px;
  border: 1px solid #E3E6E8;
  box-sizing: border-box;
  border-radius: 8px;
  margin-bottom: 16px;
}

.c0 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 8px;
}

.c1 {
  color: #E51A1A;
  font-size: 20px;
  padding-bottom: 16px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  gap: 16px;
}

.c2 > *:first-child {
  -webkit-box-flex: 4;
  -webkit-flex-grow: 4;
  -ms-flex-positive: 4;
  flex-grow: 4;
}

.c2 > *:last-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<body>
  <div>
    <div
      class="c0"
    >
      <h1
        class="c1"
      >
        Danger Zone
      </h1>
      <div>
        <form
          class="c2"
        >
          <div>
            <label
              class="c3"
              for="partitions"
            >
              Number of partitions *
            </label>
            <div
              class="c4"
            >
              <input
                class="c5 c4"
                id="partitions"
                name="partitions"
                placeholder="Number of partitions"
                type="number"
              />
            </div>
          </div>
          <div>
            <button
              class="c6"
              data-testid="partitionsSubmit"
              disabled=""
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
        <p
          class="c7"
        />
        <form
          class="c2"
        >
          <div>
            <label
              class="c3"
              for="replicationFactor"
            >
              Replication Factor *
            </label>
            <div
              class="c4"
            >
              <input
                class="c8 c4"
                id="replicationFactor"
                name="replicationFactor"
                placeholder="Replication Factor"
                type="number"
              />
            </div>
          </div>
          <div>
            <button
              class="c6"
              data-testid="replicationFactorSubmit"
              disabled=""
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
        <p
          class="c7"
        />
      </div>
    </div>
  </div>
</body>
`;
